### PR TITLE
Support for Redis backend repository

### PR DESCRIPTION
**Pull request by piomin**
Implementation for Spring Cloud Config Server to enable Redis backend repository

### CODE REVIEW

**Comment by pivotal-issuemaster** [comment]
@piomin Please sign the [Contributor License Agreement](https://cla.pivotal.io/sign/spring?repositoryId=spring-cloud/spring-cloud-config&pullRequestId=1282)!

[Click here](https://cla.pivotal.io/sync/spring?repositoryId=spring-cloud/spring-cloud-config&pullRequestId=1282) to manually synchronize the status of this Pull Request.

See the [FAQ](https://cla.pivotal.io/about) for frequently asked questions.

**Comment by pivotal-issuemaster** [comment]
@piomin Thank you for signing the [Contributor License Agreement](https://cla.pivotal.io/sign/spring?repositoryId=spring-cloud/spring-cloud-config&pullRequestId=1282)!

File: spring-cloud-config-server/src/main/java/org/springframework/cloud/config/server/config/EnvironmentRepositoryConfiguration.java [hunk 1]
```diff
@@ -81,12 +85,13 @@
...
 		CredhubEnvironmentProperties.class, JdbcEnvironmentProperties.class,
-		NativeEnvironmentProperties.class, VaultEnvironmentProperties.class })
+		NativeEnvironmentProperties.class, VaultEnvironmentProperties.class,
+		RedisEnvironmentProperties.class })
 @Import({ CompositeRepositoryConfiguration.class, JdbcRepositoryConfiguration.class,
 		VaultRepositoryConfiguration.class, CredhubConfiguration.class,
 		CredhubRepositoryConfiguration.class, SvnRepositoryConfiguration.class,
 		NativeRepositoryConfiguration.class, GitRepositoryConfiguration.class,
-		DefaultRepositoryConfiguration.class })
+		DefaultRepositoryConfiguration.class, RedisRepositoryConfiguration.class })
```
Comment (spencergibb) [inline spring-cloud-config-server/src/main/java/org/springframework/cloud/config/server/config/EnvironmentRepositoryConfiguration.java]: Default needs to be last

File: spring-cloud-config-server/src/main/java/org/springframework/cloud/config/server/environment/VaultEnvironmentProperties.java [hunk 2]
```diff
@@ -74,7 +74,8 @@
 	private int kvVersion = 1;
 
 	/**
-	 * The value of the Vault X-Vault-Namespace header. Defaults to null. This a Vault Enterprise feature only.
+	 * The value of the Vault X-Vault-Namespace header. Defaults to null. This a Vault
```
Comment (spencergibb) [inline spring-cloud-config-server/src/main/java/org/springframework/cloud/config/server/environment/VaultEnvironmentProperties.java]: Please revert

File: spring-cloud-config-server/src/main/java/org/springframework/cloud/config/server/config/EnvironmentRepositoryConfiguration.java (hunk 1 above)
Comment (piomin) [reply spring-cloud-config-server/src/main/java/org/springframework/cloud/config/server/config/EnvironmentRepositoryConfiguration.java]: Done

File: spring-cloud-config-server/src/main/java/org/springframework/cloud/config/server/environment/VaultEnvironmentProperties.java (hunk 2 above)
Comment (piomin) [reply spring-cloud-config-server/src/main/java/org/springframework/cloud/config/server/environment/VaultEnvironmentProperties.java]: Reverted

**Comment by piomin** [comment]
@ryanjbaxter i have already fixed errors caused by my changes. There are still two errors during build, but I seems they are not related with my changes.
Where should I commit documentation?

**Comment by ryanjbaxter** [comment]
Documentation can be added here https://github.com/spring-cloud/spring-cloud-config/blob/master/docs/src/main/asciidoc/spring-cloud-config.adoc

I built your branch locally and it also fails for me, so the failures have something to do with your changes.

**Comment by piomin** [comment]
Ok, you are right. The problem in this test is with `spring-data-redis` dependency. I can exclude it together with `spring-jdbc`, and it finishes with success, but I'm not sure if it is right solution
```
@RunWith(ModifiedClassPathRunner.class)
@ClassPathExclusions({"spring-jdbc-*.jar", "spring-data-redis-*.jar"})
public static class JdbcTests
```

**Comment by ryanjbaxter** [comment]
What is the problem with them being included, is there some kind of autoconfiguration that is happening?

**Comment by spencergibb** [comment]
That's my suspicion.

**Comment by piomin** [comment]
@ryanbaxter I have already update the documentation. 
I'm not sure what is the purpose of that test. I think that spring-data-redis includes some dependencies that may broke that test.

File: docs/src/main/asciidoc/spring-cloud-config.adoc [hunk 3]
```diff
@@ -757,6 +757,52 @@ The database needs to have a table called `PROPERTIES` with columns called `APPL
...
+----
+HGETALL sample-app
+{
+  "server.port": "8100",
+  "sample.topic.name": "test",
+  "test.property1": "property1"
+}
+----
+
+NOTE: When no profile is specified `default` will be used
```
Comment (spencergibb) [inline docs/src/main/asciidoc/spring-cloud-config.adoc]: Missing punctuation.

Comment (piomin) [reply docs/src/main/asciidoc/spring-cloud-config.adoc]: Ok, which line?

Comment (spencergibb) [reply docs/src/main/asciidoc/spring-cloud-config.adoc]: That line is missing a period at the end of the sentence.

Comment (piomin) [reply docs/src/main/asciidoc/spring-cloud-config.adoc]: fixed

**Pull request merged**